lib/guardian_db.ex

```diff
@@ -81,26 +82,43 @@ defmodule GuardianDb do
   end
 
   @doc """
-  After the JWT is generated, stores the various fields of it in the DB for tracking
+  After the JWT is generated, stores the various fields of it in the DB for tracking if
+  token type matches the configured types to be stored.
   """
   def after_encode_and_sign(resource, type, claims, jwt) do
-    case Token.create!(claims, jwt) do
+    case store_token(type, claims, jwt) do
       {:error, _} -> {:error, :token_storage_failure}
       _           -> {:ok, {resource, type, claims, jwt}}
     end
   end
 
+  defp store_token(type, claims, jwt) do
+    if storable_type?(type) do
+      Token.create!(claims, jwt)
+    else
+      :non_storable_type
+    end
+  end
+
   @doc """
-  When a token is verified, check to make sure that it is present in the DB.
+  When a token is verified, check to make sure that it is present in the DB if it matches the configured token storage types.
   If the token is found, the verification continues, if not an error is returned.
   """
   def on_verify(claims, jwt) do
-    case Token.find_by_claims(claims) do
+    case find_token(claims) do
       nil    -> {:error, :token_not_found}
       _token -> {:ok, {claims, jwt}}
     end
   end
 
+  defp find_token(%{"typ" => type} = claims) do
+    if storable_type?(String.to_existing_atom(type)) do
+      Token.find_by_claims(claims)
+    else
+      :ok
```

Comment:
If you remove this `else` completely, then this function will return `nil` if `storable_type?/1` returns `false`. Then the calling function will match on `nil` as it is already.

Reply:
This is another hook that should just pass through if the token type is not configured to be stored. The `nil` case should only return if a token of the expected type is not found.